From f2f4fa1ecea43bc1456d824668a1b56377099d9a Mon Sep 17 00:00:00 2001
From: Ren Chenglei <chenglei.ren@intel.com>
Date: Fri, 9 Jun 2023 21:41:20 +0530
Subject: [PATCH] INTERNAL: Fix CtsNNAPITestCases failures

For DRM_FORMAT_R8, we don't need align the stride. This
change could help fix allocator issues in CtsNNAPITestCases.

Tracked-On: OAM-95724
Signed-off-by: Ren Chenglei <chenglei.ren@intel.com>
---
 i915.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/i915.c b/i915.c
index 2aa04e9..b6ca8a1 100644
--- a/i915.c
+++ b/i915.c
@@ -409,6 +409,9 @@ static int i915_align_dimensions(struct bo *bo, uint32_t format, uint32_t tiling
 
 	*aligned_height = ALIGN(*aligned_height, vertical_alignment);
 	if (i915->graphics_version > 3) {
+#ifdef USE_GRALLOC1
+		if (DRM_FORMAT_R8 != bo->meta.format)
+#endif
 		*stride = ALIGN(*stride, horizontal_alignment);
 	} else {
 		while (*stride > horizontal_alignment)
-- 
2.40.1

